diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -331,100 +361,31 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
-            prevNode = node
-            node = node.treeParent
-            if (node.elementType == PARENTHESIZED) {
-                text += getTextFromParenthesized(node)
             }
-        } while (node.elementType != BINARY_EXPRESSION)
-
-        if (node.firstChildNode == prevNode) {
-            if (node.treePrev != null && node.treePrev.elementType == WHITE_SPACE) {
-                text += node.treePrev.text
-            }
-        } else {
-            if (prevNode.treePrev != null && prevNode.treePrev.elementType == WHITE_SPACE) {
-                text += prevNode.treePrev.text
+            if (!nextNode.text.contains(("\n"))) {
+                rigthSplitnode.first.appendNewlineMergingWhiteSpace(nextNode, nextNode)
             }
         }
-        while (node.treeParent.elementType == PARENTHESIZED) {
-            node = node.treeParent
-            text += getBraceAndBeforeText(node, prevNode)
-        }
-        text += astNode.text
-        node = astNode.parent({ newNode -> newNode.nextSibling { it.elementType == OPERATION_REFERENCE } != null },
-            strict = false)
-            ?: return text
-        node = node.nextSibling { it.elementType == OPERATION_REFERENCE }!!
-        if (node.treePrev.elementType == WHITE_SPACE) {
-            text += node.treePrev.text
-        }
-        text += node.text
-        return text
-    }
-
-    @Suppress("UnsafeCallOnNullableType")
-    private fun getBraceAndBeforeText(node: ASTNode, prevNode: ASTNode): String {
-        val par = prevNode.prevSibling { it.elementType == OPERATION_REFERENCE }?.let { LPAR } ?: RPAR
-        var text = ""
-        if (node.findChildByType(par)!!.treePrev != null &&
-                node.findChildByType(par)!!.treePrev.elementType == WHITE_SPACE) {
-            text += node.findChildByType(par)!!.treePrev.text
-        }
-        text += node.findChildByType(par)!!.text
-        return text
-    }
-
-    @Suppress("UnsafeCallOnNullableType")
-    private fun getTextFromParenthesized(node: ASTNode): String {
-        var text = ""
-        text += node.findChildByType(LPAR)!!.text
-        if (node.findChildByType(LPAR)!!.treeNext.elementType == WHITE_SPACE) {
-            text += node.findChildByType(LPAR)!!.treeNext.text
-        }
-        if (node.findChildByType(RPAR)!!.treePrev.elementType == WHITE_SPACE) {
-            text += node.findChildByType(RPAR)!!.treePrev.text
-        }
-        text += node.findChildByType(RPAR)!!.text
-        return text
     }
 
     /**
-     * This method stored all the nodes that have BINARY_EXPRESSION or PREFIX_EXPRESSION element type.
-     * This method uses recursion to store binary node in the order in which they are located
-     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
-     *
-     *@param node node in which to search
-     *@param binList mutable list of ASTNode to store nodes
+     * Recursively searches for all Binary expressions inside a given
```

Comment:
Or at least complete your description
```suggestion
     * Recursively searches for all Binary expressions inside a given node, and stores them into the [binList]
```

Reply:
Returned the old comments